src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -327,6 +328,62 @@ public static FormulaValue Text(EvalVisitor runner, EvalVisitorContext context,
             return Text(runner.GetFormattingInfo(), irContext, args, runner.CancellationToken);
         }
 
+        public static FormulaValue TextToBlob(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, FormulaValue[] args)
+        {
+            runner.CancellationToken.ThrowIfCancellationRequested();
+
+            if (args[0].IsBlank())
+            {
+                return new BlobValue(irContext);
+            }
+
+            if (args[0] is StringValue sv)
+            {
+                IResourceManager resourceManager = runner.FunctionServices.GetService<IResourceManager>();
+
+                if (resourceManager == null)
+                {
+                    return new ErrorValue(irContext, new ExpressionError()
+                    {
+                        Message = $"Cannot convert {args[0].Type._type} to Blob, missing resource manager.",
+                        Span = irContext.SourceContext,
+                        Kind = ErrorKind.InvalidArgument
+                    });
+                }
+
+                return new BlobValue(resourceManager, resourceManager.NewElementFromString(sv.Value));

```

Comment:
>return new BlobValue(resourceManager, resourceManager.NewElementFromString(sv.Value)); [](http://example.com/codeflow?start=16&length=86)

Can we avoid creating a new element in the resource manager?

In an expression such as `If(false, <blobexpr>, If(false, <stringexpr>, <blobvalue>))` the tree would be something like this:

    If(
        false,
        <blobexpr>
        StringToBlob(
            If(
                false,
                <stringexpr>,
                BlobToString(<blobvalue>)
            )
        )
    )

That means that we would be creating a new element there in this case, where it doesn't have to be

Reply:
Is this only an optimization you're proposing? (and I don't know how to implement it)

Reply:
Not just an optimization. If you are creating a new resource, would it be "equal" to the previous one? The value `<blobvalue>` should be equal to

    If(
        false,
        <blobexpr>
        StringToBlob(
            If(
                false,
                <stringexpr>,
                BlobToString(<blobvalue>)
            )
        )
    )

Which we can accomplish if we use the URI as an "identity" for the blob value.

Reply:
This will not work as BlobToText will not work